File: Python/Day17_14_05_2025/MultilevelInheritance.py
""" Q3. Multi-level Inheritance:
Implement a three-level inheritance hierarchy. 
Create a base class Vehicle with a method show_vehicle_info(). 
Derive a class Car that adds a method show_car_info(), and further 
derive a class SportsCar that adds a method show_sports_car_info(). 
Show how methods from all classes can be accessed from an instance of the SportsCar class."""
class Vehicle:
    def show_vehicle_info(self):
        print("This is a vehicle.")
class Car(Vehicle):
    def show_car_info(self):
        print("This is a car.")
class SportsCar(Car):
    def show_sports_car_info(self):
        print("This is a sports car.")
my_sports_car = SportsCar()
my_sports_car.show_vehicle_info()    
my_sports_car.show_car_info()           
my_sports_car.show_sports_car_info()

File: Java/Day11_14_05_2025/src/MultilevelInheritance.java
/* 2.⁠ ⁠Multilevel Inheritance
Write a program with three classes:
Vehicle with a method startEngine().
Car that extends Vehicle and adds a method drive().
SportsCar that extends Car and adds a method race().
Task: Create an object of SportsCar in the main method and demonstrate the functionality of all three methods. */
class Vehicle {
    void startEngine() {
        System.out.println("Engine started.");
    }
}
class Car extends Vehicle {
    void drive() {
        System.out.println("Car is driving.");
    }
}
class SportsCar extends Car {
    void race() {
        System.out.println("SportsCar is racing.");
    }
}
public class MultilevelInheritance {
    public static void main(String[] args) {
        SportsCar mySportsCar = new SportsCar();
        mySportsCar.startEngine(); 
        mySportsCar.drive();        
        mySportsCar.race();       
    }
}